ra_aid/agent_utils.py

```
@@ -528,44 +723,68 @@ def run_agent_with_retry(agent, prompt: str, config: dict) -> Optional[str]:
+                        error_str = str(e).lower()
+                        if "code" not in error_str or "429" not in error_str:
+                            raise  # Re-raise ValueError if it's not a Lambda 429
                     if attempt == max_retries - 1:
                         logger.error("Max retries reached, failing: %s", str(e))
-                        raise RuntimeError(f"Max retries ({max_retries}) exceeded. Last error: {e}")
-                    logger.warning("API error (attempt %d/%d): %s", attempt + 1, max_retries, str(e))
-                    delay = base_delay * (2 ** attempt)
-                    print_error(f"Encountered {e.__class__.__name__}: {e}. Retrying in {delay}s... (Attempt {attempt+1}/{max_retries})")
+                        raise RuntimeError(
+                            f"Max retries ({max_retries}) exceeded. Last error: {e}"
+                        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)